lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -235,4 +235,33 @@ sub save_results {
     path($self->job->result_dir, 'details-' . $self->name . '.json')->spurt(encode_json($results));
 }
 
+# Collect text output into JSON. save_results() is called too early for this.
+sub finalize_results {
+    my ($self) = @_;
+
+    my $dir = $self->job->result_dir();
+    return unless $dir;
+    my $file    = path($dir, "details-" . $self->name . ".json");
+    my $tmpfile = $file->sibling($file->basename . '.tmp');
+    return unless -e $file;
+    my $results;
+
+    eval {
+        $results = decode_json($file->slurp);
+        $results = {details => $results} if ref($results) eq 'ARRAY';
+
+        for my $step (@{$results->{details}}) {
+            next unless $step->{text};
+            my $txtfile = path($dir, $step->{text});
+            next unless -e $txtfile;
+            $step->{text_data} = $txtfile->slurp;
+        }
+
+        $tmpfile->spurt(encode_json($results));
+        rename $tmpfile->to_string, $file->to_string;
+    };
```

Comment:
Suppressing any errors without a trace in the logs makes debugging any (unexpected) problems very annoying. Likely the Minion job should even fail if we encounter problems any here. Trying to clean up the temp file in any case and trying to move on to the next module makes of course still sense but the overall Minion job should fail in that case and contain the error messages as result.

Reply:
I don't see the point of killing the job. The worst-case scenario here is that the original JSON file will be left untouched and loading the job detail page will be slightly slower than it should be. But I'll print any errors into the log.

Reply:
I didn't mean killing the job *immediately* when an exception occurs. I'd just take that into account at the end and report the Minion job as failed if there was at least one problem. But it isn't *that* important when the errors are at least printed into the log.